File: gluecodium/src/test/resources/smoke/inheritance/output/android/com/example/smoke/ChildInterfaceImpl.java
/*
 *
 */
package com.example.smoke;
class ChildInterfaceImpl extends ParentInterfaceImpl implements ChildInterface {
    protected ChildInterfaceImpl(final long nativeHandle, final Object dummy) {
        super(nativeHandle, dummy);
    }
    public native void childMethod();
}

File: gluecodium/src/test/resources/smoke/inheritance/output/android-kotlin/com/example/smoke/ChildInterfaceImpl.kt
/*

 *
 */

@file:JvmName("ChildInterfaceImpl")

package com.example.smoke

import com.example.NativeBase

/**
 * @suppress
 */
class ChildInterfaceImpl : NativeBase, ChildInterface {
    protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}



    override external fun childMethod() : Unit




    override external fun rootMethod() : Unit
    override var rootProperty: String
        external get
        external set


    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)
    }
}
